src/rai_core/rai/communication/ros2/connectors/ari_connector.py

```diff
@@ -39,12 +39,77 @@
 
 
 class ROS2ARIConnector(ROS2ActionMixin, ROS2ServiceMixin, ARIConnector[ROS2ARIMessage]):
+    """ROS2-specific implementation of the ARIConnector.
+
+    This connector provides functionality for ROS2 communication through topics,
+    services, and actions, as well as TF (Transform) operations.
+
+    Parameters
+    ----------
+    node_name : str, optional
+        Name of the ROS2 node. If not provided, generates a unique name with UUID.
+    destroy_subscribers : bool, optional
+        Whether to destroy subscribers after receiving a message, by default False.
+        If True, the subscriber will be destroyed after the message is received.
+        If False, may lead to performance issues due to unneded subscribers.
+        It has been observed that destroying subscribers may lead to a crash of the node/executor.
+
+    Attributes
+    ----------
+    _node : Node
+        The ROS2 node instance.
+    _topic_api : ROS2TopicAPI
+        API for handling ROS2 topic operations.
+    _service_api : ROS2ServiceAPI
+        API for handling ROS2 service operations.
+    _actions_api : ROS2ActionAPI
+        API for handling ROS2 action operations.
+    _tf_buffer : Buffer
+        Buffer for storing TF transforms.
+    tf_listener : TransformListener
+        Listener for TF transforms.
+    _executor : MultiThreadedExecutor
+        ROS2 executor for running the node.
+    _thread : Thread
+        Thread running the executor.
```

Comment:
```suggestion
    tf_listener : TransformListener
        Listener for TF transforms.
```

Protected and private Attributes should not be listed in the docstring documenting the classes public API. If any of these should become part of the public API please remove the `_` prefix.

Reply:
renamed tf_listener, to _tf_listener, as there is no need to expose the listener.
removed Attributes section of the docstring